File: umbrella-insurance-core/src/main/java/com/umbrella/insurance/core/models/entities/Division.java
package com.umbrella.insurance.core.models.entities;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

@Getter
@Setter
@Entity
@Table(name = "divisions", schema = "public", uniqueConstraints = {
        @UniqueConstraint(name = "divisions_unique", columnNames = {"division_name"})
})
public class Division {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "division_id", nullable = false)
    private Long id;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "conference_id")
    private Conference conference;

    @NotNull
    @Column(name = "division_name", nullable = false, length = Integer.MAX_VALUE)
    private String divisionName;

}

File: umbrella-insurance-react-ui/src/models/leagues/conferences/divisions/v1/Division.ts
import { Conference } from "../../v1/Conference";

export class Division {
    id?: number;
    divisionName?: string;
    conference?: Conference;
    constructor(obj?:any) {
        if(obj) {
            if(obj?.id) {
                this.id = obj.id;
            }
            if(obj?.divisionName) {
                this.divisionName = obj.divisionName;
            }
            if(obj?.conference) {
                this.conference = new Conference(obj.conference);
            }
        }
    }
}